.github/workflows/documentation.yml

```diff
@@ -1,110 +1,166 @@
 name: Documentation
 
-# Trigger the documentation build for any push, pull request and any branch only deploy on next
+# This workflow is divided in 3 jobs, ocaml_docs, sphinx_docs, deploy_docs.
+# - The ocaml_docs build the ocaml documentation, it runs for every push,
+# if it fails no more work is done
+# - The sphinx_docs job build the sphinx documentation, it runs only if a PR
+# is open. if it fails no more work is done
+# - deploy_docs only run on the next branch when code is pushed. It retrieve
+# the ocaml and sphinx documentation and deploy them on the gh-pages branch
+
 on: [push, pull_request]
 
 env:
   OCAML_DEFAULT_VERSION: 4.10.0
 
 jobs:
+  # For any push and PR, build the documentation from the ocaml comments
+  # If this build fails, the documentation workflow stops
+  # If it succed, an artifact is made with the generated documentation
+  # (html format only). This artifact is used in the deploying job
   ocaml_docs:
-    # Build the documentation from the ocaml comments
-    # If this test fails, no more test
     name: OCaml documentation
 
-    strategy:
-       matrix:
-        cache-name:
-          - odoc
-
     runs-on: ubuntu-latest
 
     steps:
+      # Checkout the code of the current branch
       - name: Checkout code
         uses: actions/checkout@v2
 
+      # Retrieve the opam cache with unique key
+      # A new cache is created/used if the `.opam` files changes or
+      # if we use another ocaml version
+      # This action only retrieve de .opam/ directory
       - name: Retrieve opam cache
         uses: actions/cache@v2
         id: cache-opam
         with:
           path: ~/.opam
-          key: v1-${{ runner.os }}-alt-ergo-${{ matrix.cache-name }}-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
+          key: v1-${{ runner.os }}-alt-ergo-odoc-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
 
+      # Get an OCaml environment with opam installed and the proper ocaml version
+      # opam will used opam cache environment if retrieved
       - name: Use OCaml ${{ env.OCAML_DEFAULT_VERSION }}
         uses: avsm/setup-ocaml@v1
         with:
           ocaml-version: ${{ env.OCAML_DEFAULT_VERSION }}
 
-      - name: Set git user
-        run: |
-          git config --global user.name github-actions
-          git config --global user.email github-actions-bot@users.noreply.github.com
-
-      - run: opam pin add . -y --no-action
-      - run: opam depext -y alt-ergo-lib alt-ergo-parsers alt-ergo altgr-ergo
-        # if: steps.cache-opam.outputs.cache-hit != 'true'
-      - run: opam install -y ./*.opam --deps-only --with-test
-        # if: steps.cache-opam.outputs.cache-hit != 'true'
+      # Add OPAMYES=true to the environment, this is usefill to replace `-y` option
+      # in any opam call
+      - name: Setup opam variables
+        run: export OPAMYES=true
```

Comment:
It's probably better to set the env as part of the env for the job (as in https://github.com/Gbury/dolmen/blob/master/.github/workflows/build.yml#L53 for instance), so that would mean putting the following right after the `runs-on` setting:
```
env:
  OPAMYES: 'true'
```